### DISCUSSION TITLE
Re-mount and children in solid-js

### DISCUSSION THREAD

**Discussion by AntonyKarachanskySS** [solidjs/solid#1150] [Q&A] [answered]
Hello, react-developer here, have 2 questions:

1. Is there something like "key" in solid-js?
Looking for re-mount capability

```jsx
const UserWidget = () => {
  // ...
  const userData = useUserData({userId});
  // ...
  
  return <UserForm key={userData.id} data={userData} />
}
```

and `UserForm` uses `useState` for initial state
so convenient option was having `key`, so when we change userId, we are sure we have "fresh start"

haven't found anything like this in solid

2. Children-templates

when making layout components we can use something like this

```jsx

const LayoutManager = ({children} => {
   // ...

   return React.Children.map(children, child => {
     if (child.type === Header) // ...
     if (child.type === Content) // ...
     if (child.type === Footer) // ...
   })
}

// so usage will look like this: 

<LayoutManager>
   <Header />
   <Content />
   <Footer />
</LayoutManager>
```

can we achieve something similar?

**Comment by ryansolid** [comment, marked as answer]
1. For sure.. `<Show>` doubles as this when you use the callback form

const UserWidget = () => {
  // ...
  const userData = useUserData({userId});
  // ...
  
  return <Show when={userData()}>{userData => <UserForm data={userData} />}</Show>
}

Remember to pass the argument through so Show can differentiate it as a callback from an accessor.

2. Yeah.. The array handling is something I want to improve to force always array as a single child isn't an array.

Regardless this thread is probably relevant: https://github.com/solidjs/solid/discussions/1141

**Comment by Blankeos** [comment]
Hi, I know this is old. 

I was here for the "Remount" (1).

But read about "Children Templates" question (2). And it looked awfully similar to a solution I wrote a blog post on: https://carlo.vercel.app/blog/how-to-dynamically-wrap-components-in-solidjs#so-how-do-we-do-it. Sharing, since it might be useful to check! Edit: Oh wait lol it's completely irrelevant.
